arch/arm/src/armv8-r/arm_head.S

```diff
@@ -325,7 +330,39 @@ arm_data_initialize:
 #endif
 
 /***************************************************************************
- * Name: arm_data_initialize
+ * Name: hsctlr_initialize
+ ***************************************************************************/
+
+	.global	hsctlr_initialize
+	.type	hsctlr_initialize, #function
+
+hsctlr_initialize:
+	mrc		CP15_HSCTLR(r0)		/* Get Hyp System Control Register */
+
+#if !defined(CONFIG_ARMV8R_DCACHE_DISABLE) && !defined(CONFIG_SMP)
```

Comment:
why check CONFIG_SMP

Reply:
In case of SMP,  cache should be enabled  after cache coherency unit is inited in later boot stage.

However,  typically ARMv8-R CPUs (cortex-r5/cortex-r52) do not have cache coherency,  there is no cache coherency unit. So smp can not be supported in those cpus.

Reply:
but R82 support the cache coherence:
https://developer.arm.com/Processors/Cortex-R82

Reply:
Yes, CONFIG_SMP checking should be removed. 

In future optimizations, I can add a configuration option for cache coherency. Different CPU architectures can choose whether they have cache coherency. If a CPU does not have cache coherency and SMP is enabled, the cache should not be enabled.